doc/source/feature_overview.md

```diff
@@ -0,0 +1,263 @@
+```
+
+## The core {class}`~geoutils.Raster` and {class}`~geoutils.Vector` classes
+
+In GeoUtils, geospatial handling is object-based and revolves around {class}`~geoutils.Raster` and {class}`~geoutils.Vector`.
+These link to either **on-disk** or **in-memory** datasets, opened by instantiating the class with a file path.
+
+```{code-cell} ipython3
+import geoutils as gu
+
+# Examples files: infrared band of Landsat and glacier outlines
+filename_rast = gu.examples.get_path("everest_landsat_b4")
+filename_vect = gu.examples.get_path("everest_rgi_outlines")
+
+# Open files by instantiating Raster and Vector
+rast = gu.Raster(filename_rast)
+vect = gu.Vector(filename_vect)
+```
+
+A {class}`~geoutils.Raster` is a composition class with four main attributes: a {class}`numpy.ma.MaskedArray` as {attr}`~geoutils.Raster.data`, a
+{class}`pyproj.crs.CRS` as {attr}`~geoutils.Raster.crs`, an [{class}`affine.Affine`](https://rasterio.readthedocs.io/en/stable/topics/migrating-to-v1.html#affine-affine-vs-gdal-style-geotransforms)
+as {attr}`~geoutils.Raster.transform`, and a {class}`float` or {class}`int` as {attr}`~geoutils.Raster.nodata`.
+
+
+```{code-cell} ipython3
+# The opened raster
+rast
+```
+
+```{important}
+When a file exists on disk, {class}`~geoutils.Raster` is linked to a {class}`rasterio.io.DatasetReader` object for loading the metadata. The array will be
+**loaded in-memory implicitly** when {attr}`~geoutils.Raster.data` is required by an operation.
+
+See {ref}`core-lazy-load` for more details.
+```
+
+A {class}`~geoutils.Vector` is a composition class with a single main attribute: a {class}`~geopandas.GeoDataFrame` as {attr}`~geoutils.Vector.ds`, for which
+most methods are wrapped directly into {class}`~geoutils.Vector`.
+
+```{code-cell} ipython3
+# The opened vector
+vect
+```
+
+All other attributes are derivatives of those main attributes, or of the filename on disk. Attributes of {class}`~geoutils.Raster` and
+{class}`~geoutils.Vector` update with geospatial operations on themselves.
+
+
+## Handling and match-reference
+
+In GeoUtils, geospatial handling operations are based on class methods, such as {func}`~geoutils.Raster.crop` or {func}`~geoutils.Raster.reproject`.
+
+For convenience and consistency, nearly all of these methods can be passed solely another {class}`~geoutils.Raster` or {class}`~geoutils.Vector` as a
+**reference to match** during the operation. A **reference {class}`~geoutils.Vector`** enforces a matching of {attr}`~geoutils.Vector.bounds` and/or
+{attr}`~geoutils.Vector.crs`, while a **reference {class}`~geoutils.Raster`** can also enforce a matching of {attr}`~geoutils.Raster.res`, depending on the nature of the operation.
+
+
+```{code-cell} ipython3
+# Print initial bounds of the raster
+print(rast.bounds)
+# Crop raster to vector's extent
+rast = rast.crop(vect)
+# Print bounds of cropped raster
+print(rast.bounds)
```

Comment:
This is not a very useful example, as the bounds don't change...
Would the reverse (vector cropped onto raster) yield a different bound?

Reply:
Yes, I did this!